### PR TITLE
Prize route and model added

### LINKED ISSUE #44: Create Prize
Simple route that creates a Prize and adds to our overall prize pool.

### LINKED ISSUE #45: Prize Model
Create prize model: consists of Name (string) and Description (string). 

For example:
{ 
    name: "25% Discount at Sweetheart Boba",
    description: "1DA19JKD"
}

### PR DESCRIPTION
**Pull request by anjanbharadwaj**
# Contributors
@anjanbharadwaj 

# Relevant issue
#44 #45 

# Summary of change
Added prize model & prize creation route.

# Testing/Verification
Passed tests
```
curl --location --request POST 'localhost:8080/prize' \
-H 'Content-Type: application/json' \
--data-raw '{
    "name": "free drift hoodie",
	"description": "pick up your merch by calling 408-408-4084"
}'
```

### CODE REVIEW

File: routes/prize_route.go
Comment (brandonLi8) [inline routes/prize_route.go]: nit: routes don't have verbs, and "create" is redundant with "post". Just use `POST("/")` here

File: controllers/prize_controller_test.go
Comment (brandonLi8) [inline controllers/prize_controller_test.go]: This function is not used. Remove

File: controllers/prize_controller_test.go
Comment (brandonLi8) [inline controllers/prize_controller_test.go]: nit: excess newline

File: controllers/prize_controller_test.go
Comment (brandonLi8) [inline controllers/prize_controller_test.go]: nit: excess newline

File: controllers/prize_controller.go
Comment (brandonLi8) [inline controllers/prize_controller.go]: JSON keys are normally lowercase, so "prize" instead of "Prize"

File: routes/prize_route.go
Comment (brandonLi8) [inline routes/prize_route.go]: Move this comment to above `g.POST("/create)` (rather than above the prizeRoute function)

Comment (brandonLi8) [inline routes/prize_route.go]: I think we should still use auth middleware, since an admin will still pass auth. In the future, what we can do is create another middleware that checks that the authenticated user is an admin. But i think its unsafe to leave out the auth middleware here.

File: controllers/prize_controller.go
Comment (brandonLi8) [inline controllers/prize_controller.go:21]: Do we want to check for uniqueness of prizes? Presumably, we wouldn't want them to add a duplicate prize by accident?

**Review by brandonLi8 — CHANGES_REQUESTED** [review]
nice work. Can you also manually test this once via curl?


Also, can you edit the `Prize` field in the Drift model to be an id instead of a string.

File: routes/prize_route.go
Comment (anjanbharadwaj) [reply routes/prize_route.go]: Sure, only downside is we'll have to unnecessarily create dummy user & login just for Prize testing even when it has no functionality right now.

File: controllers/prize_controller.go
Comment (anjanbharadwaj) [reply controllers/prize_controller.go:21]: I don't think so. As of now, we don't really have a plan as to what format the name/description fields will be. For things like coupon codes, sure maybe it makes sense, but what about other types of prizes that don't have a 'digital format' yet (ex: piece of merch), for those we'd expect name/description to be the same.

My guess is later on we'll add some field which we can use to validate

File: routes/prize_route.go
Comment (anjanbharadwaj) [reply routes/prize_route.go]: Gonna hold off on this until Aiden's PR is merged (since there are some changes for testing helpers, and I can't cherrypick one commit w/o pulling some unwanted changes).

**Comment by anjanbharadwaj** [comment]
Sure, I'll add a curl test, but also now that we have actual tests do we need to keep including these in our description? Seems a bit redundant. We should probably consider making a Postman team workspace to share manual tests there.

File: routes/prize_route.go
Comment (brandonLi8) [reply routes/prize_route.go]: > Sure, only downside is we'll have to unnecessarily create dummy user & login just for Prize testing even when it has no functionality right now.


Not sure why you have to do this? You can have it in the route, but you don't use it in the controller at all? (we don't test routes, just controllers)

Edit: clarifying: "testing" as in unit tests right?

File: controllers/prize_controller.go
Comment (brandonLi8) [reply controllers/prize_controller.go:21]: Ok, I think we should add another field (because we should display something if someone adds a duplicate prize by accident). Let's do this in a separate ticket. Can you make an issue?

**Comment by brandonLi8** [comment]
> Sure, I'll add a curl test, but also now that we have actual tests do we need to keep including these in our description? Seems a bit redundant. We should probably consider making a Postman team workspace to share manual tests there.

No, not super necessary, but still good to do it e2e once. Ask @teekenzie but i think the Postman team won't work (smth about pricing happened with our presoroban workspace).

File: routes/prize_route.go
Comment (anjanbharadwaj) [reply routes/prize_route.go]: Yep unit tests. Never mind then, forgot that tests are only looking at the controllers (though this means we might want to figure out how we can test routes as well, for the ones that are auth-protected).

Will just address this right now then & not change the test.

Comment (brandonLi8) [reply routes/prize_route.go]: > Yep unit tests. Never mind then, forgot that tests are only looking at the controllers (though this means we might want to figure out how we can test routes as well, for the ones that are auth-protected).

We decided that we are testing middle ware separately (yet to be wrritten, i'll make issues). Other than that, we don't need to test every route that is auth-protected, just the ones that use the userID from the middleware. We also decided that routes will not be tested.

File: routes/prize_route.go
Comment (brandonLi8) [inline routes/prize_route.go]: Let's change this comment now that we have auth. Just mention that we will need middleware for admin in the future.

**Review by brandonLi8 — APPROVED** [review]
good work

**Comment by teekenzie** [comment]
> No, not super necessary, but still good to do it e2e once. Ask @teekenzie but i think the Postman team won't work (smth about pricing happened with our presoroban workspace).

I am 100% down to share a Postman workspace. However, last time I checked, my Postman team was bugging out due to the expiration of my trial pro account or something like that. I am absolutely down to try creating one again though.

**Review by teekenzie — APPROVED** [review]
Looks good to me. Nice work!

**Pull request merged**
